ninja/operation.py

```diff
@@ -147,7 +150,12 @@ def _run_checks(self, request: HttpRequest) -> Optional[HttpResponse]:
     def _run_authentication(self, request: HttpRequest) -> Optional[HttpResponse]:
         for callback in self.auth_callbacks:
             try:
-                result = callback(request)
+                if asyncio.iscoroutinefunction(callback):
+                    result = async_to_sync(callback)(request)
+                elif isinstance(callback, AuthBase) and asyncio.iscoroutinefunction(callback.authenticate):
+                    result = async_to_sync(callback)(request)
+                else:
+                    result = callback(request)
```

Comment:
use `asgiref.iscoroutinefunction` :) 

It provides a backport of this functionality

Reply:
@baseplate-admin 
you mean `asgiref.sync.iscoroutinefunction` ?

Reply:
Yes